### PR TITLE
Rework [WIP]

### PR DESCRIPTION
**Pull request by DeepDiver1975**
:construction: 

See open tasks for rework: https://github.com/owncloud/calendar/issues/558

### CODE REVIEW

File: 3rdparty/VObject/Component.php [hunk 1]
```diff
@@ -0,0 +1,473 @@
+<?php
+
+namespace OCA\Calendar\Sabre\VObject;
```
Comment (DeepDiver1975) [inline 3rdparty/VObject/Component.php]: @georgehrke @raghunayyar What is this about? 3rdparty with owncloud namespace? WTF?

**Comment by jancborchardt** [comment]
Wow – giant change already. And it seems to be 500 or so commits behind. @raghunayyar @georgehrke we should learn from past mistakes and not have giant rewrites (see user management).

Can we have clarification on this?

**Comment by jancborchardt** [comment]
Please divide this into blocks and smaller pull requests. Currently I can’t even add events.

The functionality of subscribing to a calendar is great, but a different feature than a rework. That could be done on the current codebase. (And would be a good and needed improvement.)

**Comment by jancborchardt** [comment]
Also @raghunayyar @georgehrke we have 17 other open pull requests in the Calendar repository: https://github.com/owncloud/calendar/pulls

That is more than in any other app, and they deserve some review. Working on this rather than reviewing these pull requests is like saying »f**\* you« to all these people – especially because their PRs will be impossible to merge after merging this here.

File: 3rdparty/VObject/Component.php (hunk 1 above)
Comment (georgehrke) [inline 3rdparty/VObject/Component.php]: the rework branch requires the latest version of sabre vobject 3.x. We will remove 3rdparty/VObject once 3rdparty has the latest version

**Comment by georgehrke** [comment]
> Please divide this into blocks and smaller pull requests. Currently I can’t even add events.
> 
> The functionality of subscribing to a calendar is great, but a different feature than a rework. That could be done on the current codebase. (And would be a good and needed improvement.)

Can we discuss this at oC conf please

**Comment by jancborchardt** [comment]
ownCloud conf is at the end of the month. GSoC ends in less than 2 weeks. Would be good have at least partial clarity before. cc @cosenal 

@georgehrke but I guess you have exams these weeks, right? Sorry then, don’t want to stress you too much. But we still need to clear this up.

**Comment by DeepDiver1975** [comment]
another scary aspect: the unit test are untouched for over one year

please read this: http://en.wikipedia.org/wiki/Test-driven_development

**Comment by georgehrke** [comment]
> another scarry aspect: the unit test are untouched for over one year

I'll have already started writing them, but due to my lack of time I only did the most import changes so raghu could continue working

**Comment by georgehrke** [comment]
> ownCloud conf is at the end of the month. GSoC ends in less than 2 weeks. Would be good have at least partial clarity before.

End of GSoC doesn't necessarily mean that the code has to be merged and ownCloud 8 won't be released before oC conf. ;)
Furthermore there are lot's of small design issues I want to fix before merging this. 

> @georgehrke but I guess you have exams these weeks, right? Sorry then, don’t want to stress you too much. But we still need to clear this up.

Next examination is Sep 1st, so I'll be free next week. Time to catch up with prs and new issues of the last weeks

**Comment by raghunayyar** [comment]
This is the most and probably the best code I have ever written, so I am very keen this goes in. GSoC doesn't affect that. ;)

**Comment by DeepDiver1975** [comment]
> This is the most and probably the best code I have ever written,

We shall prove this! :wink:

**Comment by jancborchardt** [comment]
@georgehrke @raghunayyar please join the #owncloud-calendar IRC channel I just opened. We should use that to build a team around the Calendar app and communicate better.

**Comment by georgehrke** [comment]
I am in the middle of my preparation for my math examination. Definitely won't be able to join before end of next week

**Comment by georgehrke** [comment]
btw, I would actually highly favor a calendars-dev mailing list instead of an irc channel

**Comment by jancborchardt** [comment]
Sorry but from experience mailing lists usually do not help anything to get stuff done. IRC channels are much more focused.

**Comment by raghunayyar** [comment]
@jancborchardt I would also prefer a ML as I am not always around on IRC considering my thesis project going on at the moment but very quick to reply to emails.

**Comment by jancborchardt** [comment]
No, mailing list is just not going to work. If we go with mailing list, we might as well just keep using Github issues only. We need the IRC channel for direct exchange, and for directly seeing who is involved and be able to speak to them.

**Comment by PVince81** [comment]
How far from completion is the rework code ?

How about putting merging it to "master" which would represent the new version (v2 ?) and have bugfixes be put on stable7 instead.

This would avoid having a huge PR on a side-branch that is less visible to all.

**Comment by PVince81** [comment]
I see there are lots of commits here just to fix bugs and colors.

If the rework code was on master, such bugs could be fixed separately (even by external contributors).
And bug reports need to specify whether it happens on stable7 (the current version) or on master (the new version).

Only trouble is if master is not ready for OC 8. In that case stable8 might need to be branched off stable7...

**Comment by georgehrke** [comment]
@PVince81 here is a list of open tasks: https://github.com/owncloud/calendar/issues/558

> How about putting merging it to "master" which would represent the new version (v2 ?) and have bugfixes be put on stable7 instead.

There is no schedule and feature freeze date for oC 8 yet. I am really eager to get this in with oC 8, but a new semester just started and I have to see when I find some leisure time for the calendar.

> I see there are lots of commits here just to fix bugs and colors.

I'll have to cleanup and squash commits

**Comment by jancborchardt** [comment]
@georgehrke @raghunayyar this is not mergeable – rebase please.

**Comment by georgehrke** [comment]
> @georgehrke @raghunayyar this is not mergeable – rebase please.

I don't see a real reason to rebase. The rework has a completely different codebase. It's like trying to rebase gallery plus onto gallery.

My plan was not to merge rework onto master, but rather rename master to master_legecy and rework to master.

@DeepDiver1975 any opinion?

**Comment by DeepDiver1975** [comment]
> My plan was not to merge rework onto master, but rather rename master to master_legecy and rework to master.

fine by me - most important question then is: do we miss any features?

**Comment by georgehrke** [comment]
> do we miss any features?

to my knowledge only the ones in https://github.com/owncloud/calendar/issues/558

**Comment by jancborchardt** [comment]
@georgehrke ok, fine if @DeepDiver1975 says it’s fine. But then please call it »master-legacy« instead of »master_legecy«. ;)

**Comment by DeepDiver1975** [comment]
@georgehrke @raghunayyar guys - please fix travis - this is a pain to watch

**Comment by georgehrke** [comment]
> please fix travis

Will do first thing when I find time for rework again, right now my highest priority is getting the calendar to work with 8.1

**Comment by DeepDiver1975** [comment]
continues to live in owncloud/calendar-rework

**Pull request closed without merging**